src/arrayMethodJoin.js

```diff
@@ -4,8 +4,28 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+    const len = this.length;
+    let joinString = '';
+
+    if (len !== 0) {
+      for (let i = 0; i < len; i++) {
+        let element = this[i];
+
+        if (element === undefined || element === null) {
+          element = '';
+        }
+
+        if (i !== len - 1) {
+          joinString += `${element}${separator}`;
+        } else {
+          joinString += `${element}`;
```

Comment:
```suggestion
        if (element !== undefined || element !== null) {
          joinString += element;
        }

        if (i !== len - 1) {
          joinString += separator;
        }
```
Watch here how your code may be more simple and shorter. In such case you will have only 2 checks instead of  3.
First, you check if `this[i]` is not null/undefined, and if so - add the element to string. If it's null or undefined - will happen nothing.
Then you check if element is no the last one in the array, and if so - you add the separator. If not - do nothing